### PR TITLE
feat: add logging

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- no log files!  gotta have log files.

# solution

- add them, and ipc for renderer logging support

### CODE REVIEW

**Comment by cdaringe** [comment]
imperfect.  tired & sleepy time

File: src/interfaces.ts [hunk 1]
```diff
@@ -5,6 +5,15 @@ import execa from 'execa'
...
 
+export type ProcessName = 'main' | 'renderer'
+export type Level = 'info' | 'warning' | 'error' | 'verbose'
+export type LogMsg = {
+  level: Level
+  tags?: string[]
+  message: string
+}
+export type ProcessLogger = (log: LogMsg & { processName: ProcessName }) => void
+export type Logger = (log: LogMsg) => void
```
Comment (swashcap) [inline src/interfaces.ts]: :+1:

File: src/bin.ts [hunk 2]
```diff
@@ -1,50 +1,63 @@
...
+    log({ level: 'verbose', message: `event "${eventName}" triggered` })
     mb.on(eventName, () => reloadConfig(appState))
   })
   reloadConfig(appState)
-  ipcMain.on('bus', (_, msg: FromUi) => {
+  ipcMain.on('bus', (_, msg: FromUi, payload: any) => {
     switch (msg) {
       case FromUi.REQUEST_OPEN_CONFIG_FOLDER:
         return edit(electron)
+      case FromUi.LOG:
```
Comment (swashcap) [inline src/bin.ts]: Might be nice to add a `{ tags: ['process-render', ...payload.tags], ...payload }` to this

Comment (cdaringe) [reply src/bin.ts]: Render is added, I just controlled that in the Singleton logger from the render side.  Maybe should have done that here?

**Pull request merged**